ps_checkout.php

```diff
@@ -980,30 +980,31 @@ public function hookActionFrontControllerSetMedia()
      */
     public function addCheckboxCarrierRestrictionsForModule(array $shopsList = [])
     {
-        if (!$shopsList) {
+        if (empty($shopsList)) {
             $shopsList = Shop::getShops(true, null, true);
         }
 
         $carriersList = Carrier::getCarriers((int) Context::getContext()->language->id, false, false, false, null, Carrier::ALL_CARRIERS);
-        $allCarriers = [];
-
-        foreach ($carriersList as $carrier) {
-            $allCarriers[] = $carrier['id_reference'];
-        }
+        $allCarriers = array_column($carriersList, 'id_reference');
+        $dataToInsert = [];
 
         foreach ($shopsList as $idShop) {
             foreach ($allCarriers as $idCarrier) {
-                $addModuleInCarrier = Db::getInstance()->execute('INSERT IGNORE
```

Comment:
There, I think we used `execute` to be able to use `INSERT IGNORE`.
Are you sure there won't be any issue by removing that ?

Reply:
I still use INSERT IGNORE 👍
See parameters of `Db::getInstance()->insert()` ;)

Reply:
That's indeed covered.